packages/@react-spectrum/s2/src/SegmentedControl.tsx

```diff
@@ -0,0 +1,309 @@
+  useLayoutEffect(() => {
+    if (domRef.current) {
+      let slide = document.querySelector('#animate');
+      let item = domRef.current.querySelector('label[data-selected=true]');
+      // let cachedItem = itemRef?.current?.getBoundingClientRect();
+
+
+      if (slide && itemRef.current) {
+        // let finalItem = item.getBoundingClientRect();
+
+        // Invert: determine the delta between the first and last bounds of the element
+        // let deltaX = cachedItem.left - finalItem.left;
+        // let deltaW = cachedItem.width / finalItem.width;
+        // let deltaW = finalItem.width / cachedItem.width;
+        // let deltaH = cachedItem.height / finalItem.height;
+        // let cachedX = itemRef?.current.offsetLeft;
+        let finalX = item.offsetLeft;
+        // let finalW = item.offsetWidth;
+        // let finalH = item.offsetHeight;
+
+        // styleObj.width = finalW;
+        // styleObj.height = finalH;
+        // setStyle(styleObj);
+        
+        // Play: animate the final element from its first bounds to its last bound
+        slide.animate(
+          [
+            {transform: `translateX(${finalX}px)`}
+          ],
+          {
+            duration: 150,
+            easing: 'ease-in',
+            fill: 'forwards'
+          }
+        );
+      }
+    }
+  }, [state?.selectedValue]);
+
+
+  // style={{transform: move ? `translateX(100px)` : 'translateX(0px)'}}
+  // style={{transform: 'translateX(120px)'}}
+  return (
+    <Provider
+      values={[
+        [InternalSegmentedControlContext, {register: register}]
+      ]}>
+      {props.children}
+      <span style={style} id="animate" className={test} /> 
+    </Provider>
+  );
+}
+
+const test = style({
+  backgroundColor: 'gray-25',
+  width: 56,
+  height: 32,
+  position: 'absolute',
+  boxSizing: 'border-box',
+  borderStyle: 'solid',
+  borderWidth: 2,
+  borderColor: 'gray-900',
+  borderRadius: 'lg',
+  zIndex: 1
+});
+
+function ControlItem(props: ControlItemProps, ref: FocusableRef<HTMLLabelElement>) {
+  let inputRef = useRef<HTMLInputElement>(null);
+  let domRef = useFocusableRef(ref, inputRef);
+  let {register} = useContext(InternalSegmentedControlContext);
+  // @ts-ignore Property 'isDisabled' does not exist on type 'RadioGroupState | null', not sure why it's complaining...
+  let {isDisabled: isRadioGroupDisabled} = useContext(RadioGroupStateContext);
+
+  useEffect(() => {
+    if (!props.isDisabled && !isRadioGroupDisabled) {
+      register?.(props.value);
+    }
+  }, []);
+
+  return (
+    <Radio 
+      {...props} 
+      ref={domRef} 
+      inputRef={inputRef}
+      style={props.UNSAFE_style}
+      className={renderProps => (props.UNSAFE_className || '') + controlItem({...renderProps}, props.styles)} >
+      <Provider 
+        values={[
+          [IconContext, {
+            render: centerBaseline({slot: 'icon', styles: style({order: 0, flexShrink: 0})})
+          }]
+        ]}>
+        {props.children}
```

Comment:
hmm not sure if i can do that without throwing a TS error since it excepts a slot prop because of the RAC implementation. it expects either "description" or "error message" which i don't think we want to add

Reply:
ah, that's an interesting case to bring up
Can you override that slot inside the RadioGroup to null it out?